src/StaticWebAssetsSdk/Tasks/ApplyCompressionNegotiation.cs

```diff
@@ -169,11 +110,11 @@ public override bool Execute()
         // We now have only endpoints that might have the same route but point to different assets
         // and we want to include them in the updated endpoints so that we don't incorrectly remove
         // them from the associated item group when we update the endpoints.
-        var endpointsByRoute = endpointsByAsset.Values.SelectMany(e => e).GroupBy(e => e.Route).ToDictionary(g => g.Key, g => g.ToList());
-
-        var updatedEndpointsByRoute = updatedEndpoints.Select(e => e.Route).ToArray();
```

Comment:
Avoid computing the routes and just iterate over the updated endpoints saving the changes in a separate list